### PR TITLE
move from request to axios

### PR DESCRIPTION
**Pull request by resdenia**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,11 +41,10 @@
     "logzio"
   ],
   "dependencies": {
+    "axios": "0.27.2",
     "json-stringify-safe": "5.0.1",
     "lodash.assign": "4.2.0",
-    "moment": "2.29.2",
-    "request": "^2.88.0",
-    "request-promise": "^4.2.4"
+    "moment": "2.29.2"
```
Comment (Doron-Bargo) [inline package.json]: please upgrade moment to 2.29.4

**Review by Doron-Bargo — APPROVED** [review]
lgtm
just upgrade the moment package

File: lib/logzio-nodejs.js [hunk 2]
```diff
@@ -78,6 +78,17 @@ class LogzioLogger {
 
         this.protocol = protocol;
         this._setProtocol(port);
+        // if (this.compress) {
+        //     axios.defaults.headers.post['content-encoding'] = 'gzip';
+        // }
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js]: remove redundant comments

File: lib/logzio-nodejs.js [hunk 3]
```diff
@@ -78,6 +78,17 @@ class LogzioLogger {
...
+        axios.defaults.headers.post = {
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js]: better create axios instance for this class as you dont want to affect all axios clients of the user/app/code
https://axios-http.com/docs/instance

File: package.json [hunk 4]
```diff
@@ -41,11 +41,10 @@
     "logzio"
   ],
   "dependencies": {
+    "axios": "0.27.2",
     "json-stringify-safe": "5.0.1",
     "lodash.assign": "4.2.0",
-    "moment": "2.29.2",
-    "request": "^2.88.0",
-    "request-promise": "^4.2.4"
```
Comment (AlonMiz) [inline package.json:48]: 👏

File: lib/logzio-nodejs.js [hunk 5]
```diff
@@ -265,33 +276,31 @@ class LogzioLogger {
...
+        return axios.post(this.url, body)
             .then(() => {
                 this._debug(`Bulk #${bulk.id} - sent successfully`);
                 this.callback();
             })
             .catch((err) => {
                 // In rare cases server is busy
-                const errorCode = err.cause && err.cause.code;
+                const errorCode = err.code;
                 if (UNAVAILABLE_CODES.includes(errorCode)) {
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js:295]: have u been able to generate one of the UNAVAILABLE_CODES?

Comment (resdenia) [reply lib/logzio-nodejs.js:295]: Yes
Based on the test `recovers after server fails one time => Msgs are only sent once`
I'm receiving `ECONNABORTED`
Screen Shot 2022-07-12 at 10 02 47

File: lib/logzio-nodejs.js (hunk 3 above)
Comment (AlonMiz) [reply lib/logzio-nodejs.js]: this is critical AFAIU

File: lib/logzio-nodejs.js [hunk 6]
```diff
@@ -1,8 +1,9 @@
-const request = require('request-promise');
 const stringifySafe = require('json-stringify-safe');
 const assign = require('lodash.assign');
 const dgram = require('dgram');
 const zlib = require('zlib');
+const axiosIn = require('./axiosInstance');
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js]: let's be explicit
```suggestion
const axiosInstance = require('./axiosInstance');
```

File: lib/logzio-nodejs.js [hunk 7]
```diff
@@ -78,6 +79,17 @@ class LogzioLogger {
 
         this.protocol = protocol;
         this._setProtocol(port);
+        this.url = `${this.protocol}://${this.host}:${this.port}?token=${this.token}`;
+
+		this.axiosInstance = axiosIn;
+        this.axiosInstance.defaults.headers.post = {
```
Comment (AlonMiz) [inline lib/logzio-nodejs.js]: what happened to your indentation
dont u work with vscode? pls enable the linting

**Review by AlonMiz — COMMENTED** [review]
great 👍

**Pull request merged**